go/pvl/interp.go

```diff
@@ -510,62 +448,68 @@ func runScript(g proofContextExt, script *jsonw.Wrapper, startstate scriptState)
-			n++
-		}
+func stepInstruction(g proofContextExt, ins instructionT, state scriptState) (scriptState, libkb.ProofError) {
+	debugWithState(g, state, "Running instruction %v: %v", ins.Name(), ins)
+
+	var newState scriptState
+	var stepErr libkb.ProofError
+	var customErrSpec *errorT
+	switch {
+	case ins.AssertRegexMatch != nil:
+		newState, stepErr = stepAssertRegexMatch(g, *ins.AssertRegexMatch, state)
+		customErrSpec = ins.AssertRegexMatch.Error
+	case ins.AssertFindBase64 != nil:
+		newState, stepErr = stepAssertFindBase64(g, *ins.AssertFindBase64, state)
+		customErrSpec = ins.AssertFindBase64.Error
+	case ins.WhitespaceNormalize != nil:
+		newState, stepErr = stepWhitespaceNormalize(g, *ins.WhitespaceNormalize, state)
+		customErrSpec = ins.WhitespaceNormalize.Error
+	case ins.RegexCapture != nil:
+		newState, stepErr = stepRegexCapture(g, *ins.RegexCapture, state)
+		customErrSpec = ins.RegexCapture.Error
+	case ins.Fetch != nil:
+		newState, stepErr = stepFetch(g, *ins.Fetch, state)
+		customErrSpec = ins.Fetch.Error
+	case ins.SelectorJSON != nil:
+		newState, stepErr = stepSelectorJSON(g, *ins.SelectorJSON, state)
+		customErrSpec = ins.SelectorJSON.Error
+	case ins.SelectorCSS != nil:
+		newState, stepErr = stepSelectorCSS(g, *ins.SelectorCSS, state)
+		customErrSpec = ins.SelectorCSS.Error
+	case ins.TransformURL != nil:
+		newState, stepErr = stepTransformURL(g, *ins.TransformURL, state)
+		customErrSpec = ins.TransformURL.Error
+	default:
+		newState = state
+		stepErr = libkb.NewProofError(keybase1.ProofStatus_INVALID_PVL,
+			"Invalid instruction: %v", ins)
 	}
 
-	if n == 1 {
-		debugWithState(g, state, "Running instruction %v: %v", name, ins.MarshalToDebug())
-		newstate, stepErr := step(g, ins, state)
-		if stepErr != nil {
-			debugWithStateError(g, state, stepErr)
+	if stepErr != nil {
+		debugWithStateError(g, state, stepErr)
 
-			if stepErr.GetProofStatus() == keybase1.ProofStatus_INVALID_PVL {
-				return newstate, stepErr
-			}
+		// Don't rewrite invalid_pvl errors
+		if stepErr.GetProofStatus() == keybase1.ProofStatus_INVALID_PVL {
+			return newState, stepErr
+		}
 
-			customErrSpec, customErrSpecErr := extractCustomErrorSpec(g, ins)
-			if customErrSpecErr != nil {
-				// Validation should have already caught this case.
-				// So here we just allow the non-custom error through.
-				customErrSpec = customErrorSpec{
-					hasStatus: false,
-					hasDesc:   false,
-				}
-			}
-			// Replace error with custom error.
-			customErr, swap := replaceCustomError(g, state, customErrSpec, stepErr)
-			if swap {
-				stepErr = customErr
-				debugWithState(g, state, "Replacing error with custom error")
-				debugWithStateError(g, state, customErr)
-			}
+		// Replace error with custom error.
+		customErr, swap := replaceCustomError(g, state, customErrSpec, stepErr)
+		if swap {
+			stepErr = customErr
+			debugWithState(g, state, "Replacing error with custom error")
+			debugWithStateError(g, state, customErr)
```

Comment:
Would it make more sense to unconditionally assign the return value to `stepErr`, and to move the logging down into `replaceCustomError`? Then you might not need the bool.